nomos-services/chain/chain-service/src/lib.rs

```diff
@@ -1539,6 +1524,66 @@ where
 
         (cryptarchia, storage_blocks_to_remove)
     }
+
+    #[expect(
+        clippy::cognitive_complexity,
+        reason = "TODO: Address this at some point."
+    )]
+    async fn broadcast_session_updates_for_block(
+        cryptarchia: &Cryptarchia,
+        block_id: &HeaderId,
+        relays: &CryptarchiaConsensusRelays<
+            Mempool,
+            MempoolNetAdapter,
+            NetAdapter,
+            SamplingBackend,
+            Storage,
+            RuntimeServiceId,
+        >,
+        previous_sessions: Option<&HashMap<ServiceType, u64>>,
+    ) {
+        let new_sessions = match cryptarchia.active_sessions_numbers(block_id) {
+            Ok(s) => s,
+            Err(e) => {
+                error!("Could not get active session numbers for block {block_id:?}: {e}");
+                return;
+            }
+        };
+
+        for (service, new_session_number) in &new_sessions {
+            // If `previous_sessions` is provided, check if the session number has changed.
+            // Otherwise, always broadcast (for initialization).
+            let should_broadcast = previous_sessions
+                .is_none_or(|prev| prev.get(service).copied().unwrap_or(0) != *new_session_number);
```

Comment:
I guess we know this is not `None` because we called `is_none_or`, so an `expect` is probably more explicit here.

```suggestion
                .is_none_or(|prev| prev.get(service).copied().expect("") != *new_session_number);
```

Reply:
Fixed here https://github.com/logos-co/nomos/pull/1829/commits/1e471dfa03baeb4584e75bd28e17f7d02b298bc1